src/nodes.coffee

```diff
@@ -949,6 +949,28 @@ exports.StringLiteral = class StringLiteral extends Literal
     unquoted = unquoted.replace /\\n/g, '\n' if csx
     unquoted
 
+  withoutQuotesInLocationData: ->
+    endsWithNewline = @originalValue[-1..] is '\n'
+    locationData = Object.assign {}, @locationData
+    locationData.first_column          += @quote.length
+    if endsWithNewline
```

Comment:
This calculation of `last_line`/`last_column` gets a little hairy (it actually has become irrelevant to AST generation since `last_line_exclusive`/`last_column_exclusive` were introduced but still might as well be completely accurate with location data)

So I added some tests to `test/location.coffee`